### PR TITLE
[EOS-23428] Multipart Operations not working after rebase on MultipartRedesign branch.

### PR DESCRIPTION
**Pull request by SeagateChaDeepak**
# Problem Statement
_[EOS-16893](https://jts.seagate.com/browse/EOS-23428):_
_Multipart Operations (GET, PUT) not working after rebase on MultipartRedesign branch_

# Design
- New object for extended metadata was being allocated. Removed that piece of code.
- State completed is never reached so used metadatasaved state in some cleanup tasks.
- Additional bugs that were uncovered related to old_oid were fixed.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [x] http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/2023/

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by nileshgovande — APPROVED** [review]

File: server/s3_post_complete_action.cc [hunk 1]
```diff
@@ -937,13 +933,15 @@ void S3PostCompleteAction::startcleanup() {
...
+      // ACTION_TASK_ADD(S3PostCompleteAction::delete_old_object, this);
     }
+    ACTION_TASK_ADD(S3PostCompleteAction::remove_new_oid_probable_record, this);
   } else if (s3_post_complete_action_state ==
              S3PostCompleteActionState::abortedSinceValidationFailed) {
     // Abort is due to validation failures in part sizes 1..n
     ACTION_TASK_ADD(S3PostCompleteAction::mark_new_oid_for_deletion, this);
-    ACTION_TASK_ADD(S3PostCompleteAction::delete_new_object, this);
+    ACTION_TASK_ADD(S3PostCompleteAction::remove_old_oid_probable_record, this);
+    // ACTION_TASK_ADD(S3PostCompleteAction::delete_new_object, this);
```
Comment (nileshgovande) [inline server/s3_post_complete_action.cc:944]: Delete if it is concurrent delete

Comment (SeagateChaDeepak) [reply server/s3_post_complete_action.cc]: made some changes @knrajnambiar76 @nileshgovande please take a look

File: server/s3_post_complete_action.cc [hunk 2]
```diff
@@ -548,7 +549,7 @@ void S3PostCompleteAction::add_part_object_to_probable_dead_oid_list(
         // bucketing of object)
         S3CommonUtilities::size_based_bucketing_of_objects(
             ext_oid_str, part_entry[0].item_size);
-        if (is_old_object) {
+        if (!is_old_object) {
```
Comment (knrajnambiar76) [inline server/s3_post_complete_action.cc:552]: if there is old object(overwritre case)
then key is <oldid>-<newoid>

else (no overwrite) then key is newoid

File: server/s3_post_complete_action.cc [hunk 3]
```diff
@@ -998,9 +997,15 @@ void S3PostCompleteAction::mark_old_oid_for_deletion() {
 }
 
 void S3PostCompleteAction::delete_old_object() {
+
+  if (S3Option::get_instance().is_s3server_obj_delayed_del_enabled()) {
```
Comment (knrajnambiar76) [inline server/s3_post_complete_action.cc:1001]: why this check
below we have check S3Option::get_instance()->is_s3server_obj_delayed_del_enabled())

Comment (SeagateChaDeepak) [reply server/s3_post_complete_action.cc:1001]: yes, reverting :)

**Pull request merged**